tutorial/terminal/main.go

```diff
@@ -49,36 +62,54 @@ func main() {
 		UseSoftwareScroll: true,
 	})
 
-	input := make([]byte, 64)
+	// read the input
+	input := make([]byte, bufSize)
 	i := 0
 	for {
 		if uart.Buffered() > 0 {
 			data, _ := uart.ReadByte()
 
+			// check the type of data
 			switch data {
-			case 13:
-				// return key
-				uart.Write([]byte("\r\n"))
-				uart.Write([]byte("You typed: "))
-				uart.Write(input[:i])
-				uart.Write([]byte("\r\n"))
-
-				terminal.Write([]byte("\r\n"))
-				terminal.Write([]byte("You typed: "))
-				terminal.Write(input[:i])
-				terminal.Write([]byte("\r\n"))
-
+			// new line or new character, let's print the input and flush the memory
+			case CarriageReturn, NewLine:
+				writeInputAndFlush(input, i)
 				i = 0
+			// any other character, let's echo it
 			default:
 				// just echo the character
 				uart.WriteByte(data)
 				input[i] = data
 
-				terminal.WriteByte(data)
+				_ = terminal.WriteByte(data)
```

Comment:
Why all the `_ =`?

Reply:
My linter was unhappy with uncaught errors 🤷‍♀️

Reply:
it is very unreadable with all the `_` IMO :crying_cat_face:

Reply:
Fixed! :)